msal/src/main/java/com/microsoft/identity/client/internal/MsalUtils.java

```diff
@@ -259,12 +280,19 @@ public static boolean hasCustomTabRedirectActivity(@NonNull final Context contex
         for (final ResolveInfo info : resolveInfoList) {
             final ActivityInfo activityInfo = info.activityInfo;
 
-            if (activityInfo.name.equals(BrowserTabActivity.class.getName())) {
+            if (activityInfo.name.equals(BrowserTabActivity.class.getName()) &&
+                    activityInfo.packageName.equals(context.getPackageName())) {
                 hasActivity = true;
             } else {
                 // another application is listening for this url scheme, don't open
                 // Custom Tab for security reasons
-                return false;
+                Logger.warn(TAG, "Another application is listening for the URL scheme. " +
```

Comment:
nit: Another application {appName} is listening for the url Scheme

Reply:
what's the scheme?

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1357/commits/8235fb5ce92a7a74f904bd1d33cc3009167c98ff